### ISSUE TITLE
Use `Reporter` for the supervised baseline, not sklearn `LogisticRegression`

### ISSUE THREAD

**Issue by norabelrose** [EleutherAI/elk#88]
Currently the main bottleneck in the training phase is fitting the scikit-learn `LogisticRegression` models, not the actual fitting of the reporters. This is because `LogisticRegression` is done on the CPU. But we could easily replicate ~the same results by fitting `Reporter` objects with a purely supervised loss, and this should speed up the training process dramatically.
